abicheck/dumper_toolchain.py

```
@@ -371,10 +383,139 @@ def _extract_explicit_std_value(
+    ``None`` on any failure (compiler not found, times out, or rejects
+    ``-E -dM`` — e.g. an MSVC ``cl.exe``, which uses different flags
+    entirely) — a probe that can't run is simply not evidence, the same
+    fail-open convention every other best-effort toolchain probe in this
+    codebase already uses. Cached per ``(compiler_bin, lang_mode)`` pair, the
+    same cost shape as ``_clang_compiler_family``/``_clang_compiler_version``,
+    so a whole dump session pays this at most once per distinct resolved
+    compiler.
+    """
+    macro_name = "__STDC_VERSION__" if lang_mode == "c" else "__cplusplus"
+    try:
+        r = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)